internal/profiles/util_test.go

```diff
@@ -634,3 +634,126 @@ func TestFilterRulesForType(t *testing.T) {
+	}{
+		{
+			name: "A short DisplayName with whitespace",
+			profile: &minderv1.Profile{
+				Name:        "profile_name",
+				DisplayName: "",
+			},
+			existingProfileNames: []string{},
+			expected:             "profile_name",
+		},
+		{
+			name: "A short DisplayName with whitespace",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "My custom profile",
+			},
+			existingProfileNames: []string{},
+			expected:             "my_custom_profile",
+		},
+		{
+			name: "A very long DisplayName with whitespaces and more than 63 characters",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "A very long profile name that is longer than sixty three characters and will be trimmed",
+			},
+			existingProfileNames: []string{},
+			expected:             "a_very_long_profile_name_that_is_longer_than_sixty_three_charac",
+		},
+		{
+			name: "A DisplayName with special characters",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "Profile with !#$() characters",
+			},
+			existingProfileNames: []string{},
+			expected:             "profile_with_characters",
+		},
+		{
+			name: "A DisplayName with alphanumeric values",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "My 1st Profile",
+			},
+			existingProfileNames: []string{},
+			expected:             "my_1st_profile",
+		},
+		{
+			name: "A DisplayName with non-alphanumeric characters and leadning & trailing whitespaces",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "  New, Profile! 123. This is a Test Display Name with Special Characters!  ",
+			},
+			existingProfileNames: []string{},
+			expected:             "new_profile_123_this_is_a_test_display_name_with_special_charac",
+		},
+		{
+			name: "A DisplayName with Leading and trailing white spaces",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "   Leading and trailing spaces   ",
+			},
+			existingProfileNames: []string{},
+			expected:             "leading_and_trailing_spaces",
+		},
+		{
+			name: "A DisplayName with mix of upper and low case",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "UPPER CASE to lower case",
+			},
+			existingProfileNames: []string{},
+			expected:             "upper_case_to_lower_case",
+		},
+		{
+			name: "Derived profile name does not exist in the current project",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "My profile",
+			},
+			existingProfileNames: []string{"other_profile", "custom_profile"},
+			expected:             "my_profile",
+		},
+		{
+			name: "Derived profile name does exist in the current project",
+			profile: &minderv1.Profile{
+				Name:        "",
+				DisplayName: "My profile",
+			},
+			existingProfileNames: []string{"other_profile", "my_profile"},
+			expected:             "my_profile-1",
+		},
+		{
+			name: "Derived profile name does exist in the current project",
```

Comment:
There are 2 tests with the same name "Derived profile name does exist in the current project", which will make debugging difficult if one fails.